interfaces/system_key.go

```diff
@@ -378,11 +384,80 @@ func RemoveSystemKey() error {
 }
 
 func MockSystemKey(s string) func() {
-	var sk systemKey
-	err := json.Unmarshal([]byte(s), &sk)
+	sk, err := SystemKeyFromString(s)
 	if err != nil {
 		panic(err)
 	}
-	mockedSystemKey = &sk
+	mockedSystemKey = sk.(*systemKey)
 	return func() { mockedSystemKey = nil }
 }
+
+type SystemKeyMismatchAction int
+
+const (
+	SystemKeyMismatchActionUndefined SystemKeyMismatchAction = iota
+	SystemKeyMismatchActionNone
+	SystemKeyMismatchActionRegenerateProfiles
+)
+
+func (s SystemKeyMismatchAction) String() string {
+	switch s {
+	case SystemKeyMismatchActionNone:
+		return "none"
+	case SystemKeyMismatchActionRegenerateProfiles:
+		return "regenerate-profiles"
+	default:
+		return fmt.Sprintf("<undefined %d>", int(s))
+	}
+}
+
+var (
+	ErrSystemKeyMismatchVersionTooHigh = errors.New("system-key in version higher than supported")
+)
+
+// SystemKeyMismatchAdvise checks the provided and currently saved system keys
+// to device whether security profiles should be regenerated. Returns
+// ErrSystemKeyMismatchVersionTooHigh when the provided system key is newer than
+// one supported by the current process.
+func SystemKeyMismatchAdvise(maybeOther any) (SystemKeyMismatchAction, error) {
```

Comment:
Should this be the verb advise or the noun advice?

Reply:
calling a function to get an advice vs called function will advise, I'm fine either way. In https://github.com/canonical/snapd/pull/15271 we changed Advise -> Advice, OTOH we have some bits for apt glue which uses advise

Reply:
With my non-native-speaker hat I'd say `AdviseOnSystemKeyMismatch` but `SystemKeyMismatchAdvice`. I'll defer to someone who claims more authority on this.